osu.Game/Rulesets/Edit/IPositionSnapProvider.cs

```diff
@@ -1,13 +1,21 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using osu.Framework.Bindables;
+using osu.Game.Beatmaps;
 using osu.Game.Rulesets.Objects;
 using osuTK;
 
 namespace osu.Game.Rulesets.Edit
 {
     public interface IPositionSnapProvider
     {
+        /// <summary>
+        /// The spacing multiplier applied to beat snap distances.
+        /// </summary>
+        /// <seealso cref="BeatmapInfo.DistanceSpacing"/>
+        IBindable<double> DistanceSpacingMultiplier { get; }
```

Comment:
Not entirely certain this belongs to the interface. Position snap is more general than distance snap, so I imagine it may not make sense to implement this in every single ruleset (in fact mania and taiko can't ever do anything useful with this, while they generally can perform position snap, just a _different_ sort of snap).

Thoughts on splitting off `IDistanceSnapProvider : IPositionSnapProvider`, implementing it only where it makes sense (osu! and catch), and resolving it via the more-specific type where it makes sense? Haven't at all checked if it's viable. just spitballing for now.

Reply:
Also, note that if the above happens, you can potentially remove `SupportsDistanceSpacing` as a property entirely and check if a non-null `IDistanceSnapProvider` has been cached instead in order to add the distance spacing controls.

Reply:
Seems good – there are other methods in `IPositionSnapProvider` which make sense to be moved to `IDistanceSnapProvider`, but I would rather keep it there for now as to not complicate this PR. 

Have added a TODO for it at least.